addons/website/controllers/main.py

```diff
@@ -388,10 +388,11 @@ def save_session_layout_mode(self, layout_mode, view_id):
         request.session[f'website_{view_id}_layout_mode'] = layout_mode
 
     @http.route('/website/snippet/filters', type='json', auth='public', website=True)
-    def get_dynamic_filter(self, filter_id, template_key, limit=None, search_domain=None, with_sample=False):
-        dynamic_filter = request.env['website.snippet.filter'].sudo().search(
-            [('id', '=', filter_id)] + request.website.website_domain()
-        )
+    def get_dynamic_filter(
+            self, filter_id, template_key, limit=None, search_domain=None, with_sample=False,
+            context=None):
+        dynamic_filter = request.env['website.snippet.filter'].sudo().with_context(
+            context | {}).search([('id', '=', filter_id)] + request.website.website_domain())
```

Comment:
Seems very hard to read, despite being linter friendly and avoiding long lines.

If you really want to split the line:
```suggestion
    def get_dynamic_filter(
        self, filter_id, template_key, limit=None, search_domain=None, with_sample=False,
        context=None,
    ):
        domain = [('id', '=', filter_id)] + request.website.website_domain()
        filter = request.env['website.snippet.filter'].sudo().with_context(context | {}).search(domain)
```

Before (current diff of the PR):
![image](https://github.com/odoo/odoo/assets/30048408/2ed0b2de-3f1b-4e3e-9246-cf96f30ffff1)

After:
![image](https://github.com/odoo/odoo/assets/30048408/9a3c492a-6a88-41bc-9c2c-b197d6d210d9)


But actually, having to pass a new `context` key seems hacky, can't the option "just" use/select another filter template?
I don't remember the code from website_sale and filters very well, and I didn't read carefully the whole diff, but I'd have expected to not have to change anything in base implementation (website) and have the option select another template (maybe a primary inherit)